main.go

```diff
@@ -83,11 +98,22 @@ func main() {
 
 	bootstrapCount := 0
 
+	var nextBlockHash *common.Hash
 	for {
-		block, err := client.BlockByNumber(context.Background(), blockNum)
+		var block *types.Block
+		if nextBlockHash == nil {
+			block, err = client.BlockByNumber(context.Background(), blockNum)
+		} else {
+			block, err = client.BlockByHash(context.Background(), *nextBlockHash)
+		}
```

Comment:
minor optimization: because we don't have to look up the number => hash in the db